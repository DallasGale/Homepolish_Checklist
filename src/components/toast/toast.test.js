import React from 'react';
import Toast, { StyledToast } from './toast.component';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import 'jest-styled-components'

const toast = mount(<Toast />);
const styledToast = shallow(<StyledToast />)


describe('<Toast />', () => {

    it('will render without exploding!', () => {
        const tree = renderer.create(<Toast />).toJSON()
        expect(tree).toMatchSnapshot();
    })
    
    it('returns the task name', () => {
        toast.setProps({
            task: 'Task #1'
        })
        expect(toast.props().task).toBe('Task #1')
    });

    describe('when task status is set to "complete"', () => {
        it('"opacity" will be set to "1"', () => {
            styledToast.setProps({
                visible: true
            })
            expect(styledToast).toHaveStyleRule('opacity', '1');
        })
    });

    
    describe('when task status is set to "not_started"', () => {
        it('"opacity" will be set to "0"', () => {
            styledToast.setProps({
                visible: false
            })
            expect(styledToast).toHaveStyleRule('opacity', '0');
        })
    });

    

});


