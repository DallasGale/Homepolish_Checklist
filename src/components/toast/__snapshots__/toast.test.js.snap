// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> will render without exploding! 1`] = `
.c0 {
  background: #2b2b2b;
  box-shadow: 0px 4px 5px 0px rgba(0,0,0,0.18);
  border-radius: 30px;
  color: white;
  font-size: 0.9em;
  line-height: 2.5em;
  height: auto;
  opacity: 0;
  padding: 2px 30px 5px;
  position: fixed;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
  right: 20px;
  top: 5px;
  text-align: center;
  width: auto;
}

<div
  className="c0"
/>
`;
